straxen/plugins/nveto_pulse_processing.py

```diff
@@ -0,0 +1,804 @@
+    res = np.zeros(nsamples, dtype=np.float32)
+    return res, -42
+
+
+@export
+@strax.growing_result(nveto_pulses_dtype(), chunk_size=int(1e4))
+@numba.njit(cache=True, nogil=True)
+def split_pulses(records, pulses, min_split_height=25, min_split_ratio=0, _result_buffer=None):
+    """
+    Function which checks for a given pulse if the pulse should be
+    split.
+
+    Note:
+        A pulse is split at the local minimum between two maxima if
+        one the height difference between one of the maxima and the
+        minimum exceeds a certain threshold, or if
+
+    Args:
+        records (np.array):
+        pulses (np.array):
+
+    Returns:
+        np.array
+
+    Notes:
+        Function assumes same dt for all channels.
+    """
+    buffer = _result_buffer
+    offset = 0
+    record_offset = np.zeros(np.max(pulses['channel'])+1, np.int32)
+    dt = records[0]['dt']
+
+    for pulse in pulses:
+        # Get data and split pulses:
+        ch = pulse['channel']
+
+        ro = record_offset[ch]
+        data, ro = get_pulse_data(records, pulse, start_index=ro, update_pulse_edges=True)
+        record_offset[ch] = ro
+
+        edges = _split_pulse(data, min_split_height, min_split_ratio)
+        edges = edges[edges >= 0]
+
+        # Convert edges into times:
+        start_time = pulse['time']
+        edges_times = edges * dt + start_time
+        # Loop over edges and store them:
+        nedges = len(edges_times) - 1
+        for ind in range(nedges):
+            res = buffer[offset]
+            res['time'] = edges_times[ind]
+            res['endtime'] = edges_times[ind + 1]
+            res['channel'] = ch
+            if nedges - 1:
+                res['split_i'] = ind + 1
+            else:
+                res['split_i'] = ind  # 0 is reserved for events which were not split.
+            offset += 1
+            if len(buffer) == offset:
+                yield offset
+                offset = 0
+    yield offset
+
+
+@numba.njit(cache=True, nogil=True)
+def _split_pulse(data, min_height=25, min_ratio=0):
+    """
+    Function which splits the PMT pulses if ncessary.
+
+    Args:
+        data (np.array):
+
+    Keyword Args:
+        min_height (int):
+        min_ratio (float):
+
+    Returns:
+        np.array: Array containing the indicies of the pulses. The
+            rest is set to -1. The array has the same length as
+            np.diff(edges), since we do not know apprioir the number
+            of pulses.
+    """
+    res = np.ones(len(data), np.int16) * -1  # There cannot be more split points than sample
+    ind = 1
+    res[0] = 0
+    for split_point in find_split_points(data, min_height=min_height, min_ratio=min_ratio):
+        res[ind] = split_point
+        ind += 1
+    if ind == 1:
+        res[ind] = len(data)
+    return res
+
+
+# -----------------------------------------------
+# Taken from split_peaks and adopted to our needs
```

Comment:
What sort of changes were needed? Could we make a common function used by both detectors? It isn't very maintainable to copy-paste-modify business code: if someone spots a bug in code A they will likely forget to fix it in code B.

Reply:
Yes I agree. I will check what is needed. I had anyhow in mind to replace this function by the natural_breaks algorithm.